src/Basis/Constructors.jl

```diff
@@ -2,26 +2,179 @@
+
+# Returns:
+The Kosloff and Tal-Ezer conformal map derived from the inverse sine function.
+
+# Example:
+```jldoctest
+# koslofftalezertransformation conformal map
+numberquadraturepoints = 5;
+g, gprime = LFAToolkit.koslofftalezertransformation(0.95);
+
+# verify
+@assert g.(LinRange(-1,1,5)) ≈ [-1.0, -0.39494881426787537, 0.0, 0.39494881426787537, 1.0]
+
+# output
+
+```
+"""
+function koslofftalezertransformation(α)
+    g(s) = asin(α * s) / asin(α)
+    gprime(s) = α / (asin(α) * sqrt(1 - (α * s)^2))
+    g, gprime
+end
+
+"""
+```julia
+haletrefethenstriptransformation(ρ)
+```
+
+# Arguments:
+- `ρ`: sum of the semiminor and semimajor axis
+
+# Returns:
+The Hale and Trefethen strip transformation
+
+# Example:
+```jldoctest
+# haletrefethenstriptransformation conformal map
+g, gprime = haletrefethenstriptransformation(1.4);
+
+# verify
+@assert g.(LinRange(-1,1,5)) ≈ [-1.0, -0.36812132798370184, 0.0, 0.36812132798370184, 1.0]
+
+# output
+
+```
+"""
+function haletrefethenstriptransformation(ρ)
+    τ = π / log(ρ)
+    d = 0.5 + 1 / (exp(τ * π) + 1)
+    π2 = π / 2
+    # Unscaled functions of u
+    gu(u) = log(1 + exp(-τ * (π2 + u))) - log(1 + exp(-τ * (π2 - u))) + d * τ * u
+    gprimeu(u) = 1 / (exp(τ * (π2 + u)) + 1) + 1 / (exp(τ * (π2 - u)) + 1) - d
+    # Normalizing factor and scaled functions of s
+    C = 1 / gu(π / 2)
+    g(s) = C * gu(asin(s))
+    gprime(s) = -τ * C / sqrt(1 - s^2) * gprimeu(asin(s))
+    g, gprime
+end
+
+"""
+```julia
+transformquadrature(points, weights, mapping)
+```
+
+# Arguments:
+- `points`:  array of quadrature points
+- `weights`: optional array of weights to transform
+- `mapping`: choice of conformal map
+
+# Returns:
+Transformed quadrature by applying a smooth mapping = (g, gprime) from the original domain.
+
+# Example:
+```jldoctest
+# generate transformed quadrature points, weights with choice of conformal map
+points, weights = LFAToolkit.gaussquadrature(5)
+mapping = sausagetransformation(9)
+mpoints, mweights = transformquadrature(points, weights, mapping)
+
+# verify:
+wsum = sum(mweights);
+@assert wsum ≈ 2.0
+
+# output
+
+```
+"""
+function transformquadrature(points, weights = nothing, mapping = nothing)
+    if isnothing(mapping)
+        if isnothing(weights)
+            return points
+        else
+            return points, weights
+        end
+    end
+    gmap, gmapprime = mapping
+    mpoints = gmap.(points)
+    if !isnothing(weights)
+        mweights = gmapprime.(points) .* weights
+        return mpoints, mweights
+    else
+        return mpoints
+    end
+end
+
 # ------------------------------------------------------------------------------
 # utility functions for generating polynomial bases
 # ------------------------------------------------------------------------------
 
 """
 ```julia
-gaussquadrature(q)
+gaussquadrature(q, mapping)
 ```
 
-Construct a Gauss-Legendre quadrature
+Construct a Gauss-Legendre quadrature with the option of applying Conformal maps
```

Comment:
Should the C be lowercase?
```suggestion
Construct a Gauss-Legendre quadrature with the option of applying conformal maps
```